hedera-mirror-rest/viewmodel/customFeeViewModel.js

```diff
@@ -26,60 +27,58 @@ class CustomFeeViewModel {
    * @param {CustomFee} customFee
    */
   constructor(customFee) {
-    if (!customFee.amount && !customFee.royaltyDenominator) {
-      return;
-    }
-    this.all_collectors_are_exempt = customFee.allCollectorsAreExempt;
-    if (customFee.amountDenominator) {
-      // fractional fee
-      this.amount = {
-        numerator: customFee.amount,
-        denominator: customFee.amountDenominator,
-      };
-
-      this.denominating_token_id = EntityId.parse(customFee.tokenId).toString();
-      this.maximum = customFee.maximumAmount || undefined;
-      this.minimum = customFee.minimumAmount;
-      this.net_of_transfers = customFee.netOfTransfers;
-    } else if (customFee.royaltyDenominator) {
-      // royalty fee
-      this.amount = {
-        numerator: customFee.royaltyNumerator,
-        denominator: customFee.royaltyDenominator,
-      };
-
-      if (customFee.amount) {
-        this.fallback_fee = this._parseFixedFee(customFee);
-      }
-    } else {
-      // fixed fee
-      Object.assign(this, this._parseFixedFee(customFee));
-    }
-
-    this.collector_account_id = EntityId.parse(customFee.collectorAccountId, {isNullable: true}).toString();
+    this.fixed_fees = _.isNil(customFee.fixedFees) ? [] : customFee.fixedFees.map((f) => this._parseFixedFee(f));
+    this.fractional_fees = _.isNil(customFee.fractionalFees)
+      ? []
+      : customFee.fractionalFees?.map((f) => this._parseFractionalFee(f, customFee.tokenId));
+    this.royalty_fees = _.isNil(customFee.royaltyFees)
+      ? []
+      : customFee.royaltyFees?.map((f) => this._parseRoyaltyFee(f));
   }
 
-  hasFee() {
-    return !!this.amount;
-  }
-
-  isFixedFee() {
-    return this.amount.denominator === undefined;
-  }
-
-  isFractionalFee() {
-    return this.amount.denominator !== undefined && this.minimum !== undefined;
+  _parseFixedFee(fixedFee) {
+    return {
+      all_collectors_are_exempt: fixedFee.allCollectorsAreExempt ?? false,
+      amount: Number(fixedFee.amount),
```

Comment:
amount will be parsed into a number if it's safe otherwise a bigint, inside the pg driver with the custom parser we provided. Converting it to Number here is wrong since it'll be wrong for any value larger than js MAX_SAFE_INTEGER

```suggestion
      amount: fixedFee.amount,
```